### PR TITLE
Skinning weights in gltf were broken by PR#5003 (vertex remapping)

### PR DESCRIPTION
**Pull request by FlorianBorn71**
This PR fixes an oversight in this PR: https://github.com/assimp/assimp/pull/5003
that caused a regression for some skinned animations in GLTF files.
Rather than backing out the optimization from the original PR, this is a proper fix to it.
Essentially, I forgot to apply the vertex remapping to the skinning weights per vertex.

### CODE REVIEW

File: code/AssetLib/glTF2/glTF2Importer.cpp [hunk 1]
```diff
@@ -432,10 +432,10 @@ static inline bool CheckValidFacesIndices(aiFace *faces, unsigned nFaces, unsign
 #endif // ASSIMP_BUILD_DEBUG
 
 template <typename T>
-aiColor4D *GetVertexColorsForType(Ref<Accessor> input) {
+aiColor4D *GetVertexColorsForType(Ref<Accessor> input, std::vector<unsigned int> *vertexRemappingTable) {
```
Comment (FlorianBorn71) [inline code/AssetLib/glTF2/glTF2Importer.cpp:435]: unrelated bugfix

File: code/AssetLib/glTF2/glTF2Importer.cpp [hunk 2]
```diff
@@ -1100,7 +1105,7 @@ void ParseExtras(aiMetadata *metadata, const CustomExtension &extension) {
     }
 }
 
-aiNode *ImportNode(aiScene *pScene, glTF2::Asset &r, std::vector<unsigned int> &meshOffsets, glTF2::Ref<glTF2::Node> &ptr) {
+aiNode *glTF2Importer::ImportNode(glTF2::Asset &r, glTF2::Ref<glTF2::Node> &ptr) {
```
Comment (FlorianBorn71) [inline code/AssetLib/glTF2/glTF2Importer.cpp:1108]: made this a member function, it becomes a bit simpler that way

File: code/AssetLib/glTF2/glTF2Importer.cpp [hunk 3]
```diff
@@ -1028,12 +1033,12 @@ static void BuildVertexWeightMapping(Mesh::Primitive &primitive, std::vector<std
...
         indices8 = new Indices8*[attr.joint.size()];
         for (size_t j = 0; j < attr.joint.size(); ++j) {
-            attr.joint[j]->ExtractData(indices8[j]);
+            attr.joint[j]->ExtractData(indices8[j], vertexRemappingTablePtr);
         }
     } else {
         indices16 = new Indices16 *[attr.joint.size()];
         for (size_t j = 0; j < attr.joint.size(); ++j) {
-            attr.joint[j]->ExtractData(indices16[j]);
+            attr.joint[j]->ExtractData(indices16[j], vertexRemappingTablePtr);
```
Comment (FlorianBorn71) [inline code/AssetLib/glTF2/glTF2Importer.cpp:1041]: These two were the reason why the weights were broken because I did not apply the remapping

**Review by kimkulling — COMMENTED** [review]
Needs refactoring

**Comment by kimkulling** [comment]
Let's do the refactorings lateron.

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**